server/lib/resolvers/builders/knex.js

```diff
@@ -0,0 +1,60 @@
+const { VM } = require('vm2')
+const { auditLog, log } = require('../../../lib/util/logger')
+const { updateResolverMetrics } = require('../../../metrics')
+const { isEmpty } = require('lodash')
+
+function buildKnexResolver (dataSource, compiledRequestMapping, compiledResponseMapping) {
+  return function resolve (obj, args, context, info) {
+    return new Promise((resolve, reject) => {
+      const dataSourceClient = dataSource.getClient()
+      const requestTimeStart = Date.now()
+      info['dataSourceType'] = dataSource.type
+
+      const sandbox = {
+        db: dataSourceClient,
+        arguments: args,
+        parent: obj
+      }
+
+      const vm = new VM({
+        sandbox
+      })
+
+      // result is whatever is implicitly returned from the script
+      // in this case it will be the query builder object from knex
+      const query = vm.run(compiledRequestMapping)
```

Comment:
Let's add some logging here on debug level so it will provide more valuable tracing.